apps/src/blockly/googleBlocklyWrapper.js

```diff
@@ -334,6 +334,10 @@ function initializeBlocklyWrapper(blocklyInstance) {
     blocklyWrapper.editBlocks = opt_options.editBlocks;
     const workspace = blocklyWrapper.blockly_.inject(container, options);
 
+    if (!options.editBlocks) {
+      workspace.addChangeListener(Blockly.Events.disableOrphans);
```

Comment:
making sure i understand: if we don't add this listener, the `addUnusedBlocksHelpListener` callback will never be executed, so we won't show the "unused block" UI?

https://github.com/code-dot-org/code-dot-org/blob/75b86e7e6ee7745cc6df367eac72739d4229c8f2/apps/src/StudioApp.js#L490-L492

Reply:
No, the unused block UI listener is added here: https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/StudioApp.js#L490

`Blockly.Events.disableOrphans` is what makes Blockly disable blocks that aren't connected to anything: https://developers.google.com/blockly/reference/js/Blockly.Events?hl=en#.disableOrphans